lib/readline.js

```diff
@@ -570,8 +570,11 @@ function commonPrefix(strings) {
 
 Interface.prototype._wordLeft = function() {
   if (this.cursor > 0) {
+    // Reverse the string and match a word near beginning
+    // to avoid quadratic time complexity
     var leading = this.line.slice(0, this.cursor);
-    var match = leading.match(/(?:[^\w\s]+|\w+|)\s*$/);
```

Comment:
I think this regexp is also used elsewhere in this file, e.g. `_deleteWordLeft`?

Reply:
Fixed.